packages/core/src/App/Containers/SetResidenceModal/__tests__/set-residence-form.spec.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Formik } from 'formik';
+import { render, screen } from '@testing-library/react';
+import SetResidenceForm from '../set-residence-form';
+
+jest.mock('@deriv/components', () => {
+    const original_module = jest.requireActual('@deriv/components');
+    return {
+        ...original_module,
+        Autocomplete: jest.fn(() => 'mockedAutocomplete'),
+    };
+});
+describe('SetResidenceForm', () => {
+    const mock_props = {
+        class_prefix: 'set-residence',
+        errors: { residence: [] },
+        touched: { residence: '' },
+        residence_list: [],
+    };
+    it('should render the component with autocomplete input box', () => {
+        render(
+            <Formik initialValues={{}} onSubmit={() => Promise.resolve()}>
+                <SetResidenceForm {...mock_props} />
+            </Formik>
+        );
+        expect(screen.getByText('mockedAutocomplete')).toBeInTheDocument();
+    });
+
+    it('should not display the hint text if there are errors', () => {
+        render(
+            <Formik initialValues={{}} onSubmit={() => Promise.resolve()}>
+                <SetResidenceForm {...{ ...mock_props, errors: { residence: ['error1'] } }} />
```

Comment:
do we need here double spread?

Reply:
yes, it wasn't working with one..
![Screenshot 2023-08-08 at 1 41 15 PM](https://github.com/binary-com/deriv-app/assets/122449658/c36e8239-ec33-4d8a-a3c6-21b7ceb40e0e)
![Screenshot 2023-08-08 at 1 43 20 PM](https://github.com/binary-com/deriv-app/assets/122449658/847063ff-87e9-4cb7-881a-5a18f32f6689)